dev/devicelab/lib/tasks/hot_mode_tests.dart

```diff
@@ -29,59 +29,109 @@ TaskFunction createHotModeTest({ bool isPreviewDart2: false }) {
       options.add('--preview-dart-2');
     setLocalEngineOptionIfNecessary(options);
     int hotReloadCount = 0;
+    Map<String, dynamic> twoReloadsData;
+    Map<String, dynamic> freshRestartReloadsData;
     await inDirectory(flutterDirectory, () async {
       rmTree(_editedFlutterGalleryDir);
       mkdirs(_editedFlutterGalleryDir);
       recursiveCopy(flutterGalleryDir, _editedFlutterGalleryDir);
       await inDirectory(_editedFlutterGalleryDir, () async {
         if (deviceOperatingSystem == DeviceOperatingSystem.ios)
           await prepareProvisioningCertificates(_editedFlutterGalleryDir.path);
+        {
+          final Process process = await startProcess(
+              path.join(flutterDirectory.path, 'bin', 'flutter'),
+              <String>['run']..addAll(options),
+              environment: null
+          );
 
-        final Process process = await startProcess(
-          path.join(flutterDirectory.path, 'bin', 'flutter'),
-          <String>['run']..addAll(options),
-          environment: null
-        );
+          final Completer<Null> stdoutDone = new Completer<Null>();
+          final Completer<Null> stderrDone = new Completer<Null>();
+          process.stdout
+              .transform(UTF8.decoder)
+              .transform(const LineSplitter())
+              .listen((String line) {
+            if (line.contains('\] Reloaded ')) {
+              if (hotReloadCount == 0) {
+                // Update the file and reload again.
+                final File appDartSource = file(path.join(
+                    _editedFlutterGalleryDir.path, 'lib/gallery/app.dart'
+                ));
```

Comment:
Does this make an actual change to the sources of just update the timestamp to trigger a compilation? 
Having it change the code would be ideal because we want to second 'flutter run' to use the smaller incremental dill file.

Reply:
It does update the source in the `appDartSource.writeAsStringSync` below.